src/map/script.cpp

```diff
@@ -12527,19 +12525,38 @@ BUILDIN_FUNC( errormes ){
 	return SCRIPT_CMD_SUCCESS;
 }
 
-/*==========================================
- *------------------------------------------*/
+/**
+ * Attempts to catch a pet with the lure item.
+ * pet {<item_id>}
+ * catchpet {<item_id>}
+*/
 BUILDIN_FUNC(catchpet)
 {
-	int32 pet_id;
-	TBL_PC *sd;
+	map_session_data* sd = nullptr;
 
 	if( !script_rid2sd(sd) )
-		return SCRIPT_CMD_SUCCESS;
+		return SCRIPT_CMD_FAILURE;
 
-	pet_id= script_getnum(st,2);
+	int32 lure_id;
```

Comment:
```suggestion
	t_itemid lure_id;
```
Might be easier to just define it as the same type here.